plugin/src/main/java/org/opensearch/ml/processor/InferenceProcessorAttributes.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.ml.processor;
+
+import java.util.List;
+import java.util.Map;
+
+import lombok.Getter;
+import lombok.Setter;
+
+@Setter
+@Getter
+public class InferenceProcessorAttributes {
+
+    protected List<Map<String, String>> inputMaps;
+
+    protected List<Map<String, String>> outputMaps;
+
+    protected String modelId;
+    protected int maxPredictionTask;
+
+    protected Map<String, String> modelConfigMaps;
+    public static final String MODEL_ID = "model_id";
+    /**
+     * The list of maps will support for one or more prediction tasks mapping
+     * model fields on the left and document fields on the right side
+     * example input_maps
+     *"input_map": [
+     *           {
+     *             "input": "book.title"
+     *           },
+     *           {
+     *             "input": "book.text"
+     *           },
+     *         ]
+     */
+    public static final String INPUT_MAP = "input_map";
+    /**
+     * example output_maps
+     *"output_map": [
+     *           {
+     *             "response.language": "book.title_language"
```

Comment:
For input map, we are going to construct a set of parameters for model request, so it's very natural that the key will be model input parameters name, the value will be the value from some document field. I think we have no questions for this part.

For output map, we are going to put some value from the model response to a document field. So I think it's more reasonable and to use document field as key, the model response as value.

Reply:
changed to input_map-model input as key, output_map-document field as key

Reply:
https://github.com/opensearch-project/ml-commons/blob/84abae78369b943e634cc22039667ef24c173939/plugin/src/test/java/org/opensearch/ml/rest/RestMLInferenceIngestProcessorIT.java#L77-L157